packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -667,6 +671,22 @@ See documentation: ${createDocumentationLink({
     this.middleware.forEach(({ instance }) => {
       instance.started();
     });
+
+    // This is the automatic Insights middleware,
+    // added when `insights` is unset and the initial results possess `queryID`.
+    // Any user-provided middleware will be added later and override this one.
+    if (typeof this._insights === 'undefined') {
+      mainHelper.derivedHelpers[0].once('result', () => {
+        const hasAutomaticInsights = this.mainIndex
+          .getScopedResults()
+          .some(({ results }) => results?._automaticInsights === true);
```

Comment:
Since it's either `true` or `undefined`, this should suffice.

```suggestion
          .some(({ results }) => results?._automaticInsights);
```